docs/source/notebooks/sparse.md

```diff
@@ -0,0 +1,140 @@
+---
+jupyter:
+  jupytext:
+    formats: ipynb,md
+    text_representation:
+      extension: .md
+      format_name: markdown
+      format_version: '1.3'
+      jupytext_version: 1.14.0
+  kernelspec:
+    display_name: Python 3
+    language: python
+    name: python3
+---
+
+<!-- #raw -->
+<a href="https://colab.research.google.com/github/adriangb/scikeras/blob/docs-deploy/refs/heads/master/notebooks/AutoEncoders.ipynb"><img src="https://www.tensorflow.org/images/colab_logo_32px.png">Run in Google Colab</a>
+<!-- #endraw -->
+
+# Sparse Inputs
+
+
+SciKeras supports sparse inputs (`X`/features).
+You don't have to do anything special for this to work, you can just pass a sparse matrix to `fit()`.
+
+In this notebook, we'll demonstrate how this works and compare memory consumption of sparse inputs to dense inputs.
+
+
+## Setup
+
+```python
+!pip install memory_profiler
+%load_ext memory_profiler
+```
+
+```python
+import warnings
+import os
+os.environ['TF_CPP_MIN_LOG_LEVEL'] = '3'
+from tensorflow import get_logger
+get_logger().setLevel('ERROR')
+warnings.filterwarnings("ignore", message="Setting the random state for TF")
+```
+
+```python
+try:
+    import scikeras
+except ImportError:
+    !python -m pip install scikeras
+```
+
+```python
+import scipy
+import numpy as np
+from scikeras.wrappers import KerasRegressor
+from sklearn.preprocessing import OneHotEncoder
+from sklearn.pipeline import Pipeline
+from tensorflow import keras
+```
+
+## Data
+
+The dataset we'll be using is designed to demostrate a worst-case/best-case scenario for dense and sparse input features respectively.
+It consists of a single categorical feature with equal number of categories as rows.
+This means the one-hot encoded representation will require as many columns as it does rows, making it very ineffienct to store as a dense matrix but very efficient to store as a sparse matrix.
```

Comment:
I think this is a perfect example, as data with many categorical features and/or categories per feature seems like the main reason to make use of a sparse matrix. Taking that to the extreme highlights the benefit of sparse matrices well.